File: solutions/cpp/grains/1/grains.cpp
#include "grains.h"

namespace grains {
    unsigned long long square(int n){

        unsigned long long v {1};
        unsigned long long p2{1};
        n--;
        while(n){
            v += p2;
            p2 *= 2;
            n--;
        }
        return v;
    }
unsigned long long total(){
        return (square(64)-1) * 2 + 1;
    }
}  // namespace grains

File: solutions/java/grains/1/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    BigInteger grainsOnSquare(final int square) {
        // throw new UnsupportedOperationException("Delete this statement and write your own implementation.");
        if(square < 1 || square > 64)
            throw new IllegalArgumentException("square must be between 1 and 64");
        BigInteger val = BigInteger.valueOf(2);
        BigInteger nval = val.pow(square-1);
        return nval;
    }

    BigInteger grainsOnBoard() {
        return BigInteger.valueOf(2).pow(64).subtract(BigInteger.valueOf(1));
    }

}
